pingpong/ud_pingpong.c

```diff
@@ -414,18 +410,17 @@ static int client_connect(void)
 
 	ret = send_xfer(addrlen);
 	if (ret != 0)
-		goto err2;
+		goto err;
 
 	// wait for reply to know server is ready
 	ret = recv_xfer(4);
 	if (ret != 0)
-		goto err2;
+		goto err;
 
 	return 0;
 
-err2:
-	free(hints->dest_addr);
-err1:
+err:
+	free(fi->dest_addr);
```

Comment:
In the error path `fi_freeinfo` eventually gets called on `fi`. Why do we need to free `fi->dest_addr` here? Isn't it free'd in `fi_freeinfo`?